### PR TITLE
Private/devidas/bootstrap 5.5 issue

### PR DESCRIPTION
**Pull request by devidask27**
## ISSUE(S):

https://platform9.atlassian.net/browse/FT-380
## SUMMARY

Added new option to bootstrap command to configure etcd backup

## ISSUE TYPE

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that may cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## IMPACTED FEATURES/COMPONENTS:

bootstrap command

#### Manual

**bootstrap on 5.5**
Screenshot 2022-04-14 at 2 52 34 PM
**bootstrap with interval backup options**
Screenshot 2022-04-14 at 3 06 07 PM
**etcd backup added successfully**
Screenshot 2022-04-14 at 3 17 06 PM
**etcd backup disabled**
Screenshot 2022-04-14 at 3 56 35 PM

#### Reviewers

/cc @anupbarve @nikhilbandi @anmolsachan

### CODE REVIEW

File: cmd/bootstrap.go [hunk 1]
```diff
@@ -366,18 +392,74 @@ func bootstrapCmdRun(cmd *cobra.Command, args []string) {
...
+	etcdBackupPath := qbert.Storageproperties{
+		LocalPath: backupPath,
 	}
-	if isEtcdBackupDisabled {
-		etcdDefaults = qbert.EtcdBackup{}
+
+	//for etcd backup there are two types of backup options we can choose
+	if cmd.Flags().Changed("use-timestamp-backups") {
+		if !cmd.Flags().Changed("use-interval-backups") {
+			zap.S().Fatalf("Either Timestamp backup or Interval backup shoud be enabled to enable etcd backup")
```
Comment (anupbarve) [inline cmd/bootstrap.go:407]: %s/shoud/should
%s/shoud be enabled/should be passed

File: cmd/bootstrap.go [hunk 2]
```diff
@@ -366,18 +392,74 @@ func bootstrapCmdRun(cmd *cobra.Command, args []string) {
...
+					IntervalInHours:        intervalInHours,
+					MaxIntervalBackupCount: maxIntervalBackupCount,
+					StorageType:            storageType,
+					StorageProperties:      etcdBackupPath,
+				}
+			}
+		}
+	} else if !cmd.Flags().Changed("use-timestamp-backups") && cmd.Flags().Changed("use-interval-backups") {
+		if cmd.Flags().Changed("interval-in-mins") {
+			//if minutes are slected
```
Comment (anupbarve) [inline cmd/bootstrap.go:432]: %s/slected/selected

File: cmd/bootstrap.go [hunk 3]
```diff
@@ -366,18 +392,74 @@ func bootstrapCmdRun(cmd *cobra.Command, args []string) {
...
+				IntervalInHours:         intervalInHours,
+				DailyBackupTime:         dailyBackupTime,
+				MaxIntervalBackupCount:  maxIntervalBackupCount,
+				MaxTimestampBackupCount: maxTimestampBackupCount,
+				StorageType:             storageType,
+				StorageProperties:       etcdBackupPath,
+			}
+		}
+	} else {
+		//if only timestamp backup option is slected
```
Comment (anupbarve) [inline cmd/bootstrap.go:455]: %s/slected/selected

**Review by anupbarve — COMMENTED** [review]
Need corresponding changes in README as well.

File: cmd/bootstrap.go (hunk 1 above)
Comment (devidask27) [reply cmd/bootstrap.go]: Done

File: cmd/bootstrap.go (hunk 2 above)
Comment (devidask27) [reply cmd/bootstrap.go]: Done

File: cmd/bootstrap.go (hunk 3 above)
Comment (devidask27) [reply cmd/bootstrap.go]: Done

**Review by anupbarve — APPROVED** [review]

**Comment by devidask27** [comment]
**Testing on 5.4**
**1.17 cluster creation**
Screenshot 2022-04-30 at 12 50 10 PM
Screenshot 2022-04-30 at 2 33 18 PM
**1.18 cluster creation**
Screenshot 2022-04-30 at 12 33 19 PM
Screenshot 2022-04-30 at 12 36 37 PM
**1.19 cluster creation**
Screenshot 2022-04-30 at 12 27 22 PM
Screenshot 2022-04-30 at 12 35 20 PM
**1.20 cluster creation**
Screenshot 2022-04-30 at 12 40 27 PM
Screenshot 2022-04-30 at 12 44 09 PM
**1.21 cluster creation**
Screenshot 2022-04-30 at 12 40 44 PM
Screenshot 2022-04-30 at 12 44 27 PM
**Testing on 5.5**
**1.20 cluster creation**
Screenshot 2022-04-30 at 1 39 59 PM
Screenshot 2022-04-30 at 1 43 44 PM
**1.21 cluster creation**
Screenshot 2022-04-30 at 1 40 14 PM
Screenshot 2022-04-30 at 1 43 16 PM

**Comment by devidask27** [comment]
Fixed this in other PR. will add new etcd configuration options later on.

**Pull request closed without merging**